BaseLib/Algorithm.h

```diff
@@ -0,0 +1,32 @@
+/**
+ * \file
+ *
+ * \copyright
+ * Copyright (c) 2012-2018, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#pragma once
+
+#include <algorithm>
+#include "Error.h"
+
+namespace BaseLib
+{
+template <typename InputIt, typename Predicate>
+typename std::iterator_traits<InputIt>::reference findElement(
```

Comment:
Maybe rename to something like `findOrDie`?

Reply:
Btw. there is similar functionality (`getOrError` for maps) in `uniqueInsert.h`.
I suggest renaming the new function to match the old scheme and merging the functionality of Algorithm.h and uniqueInsert.h in a single file.

Reply:
#2161